core/src/main/java/org/apache/calcite/plan/RelOptRule.java

```diff
@@ -536,8 +536,8 @@ public RelTrait getOutTrait() {
    * Returns the description of this rule.
    *
    * <p>It must be unique (for rules that are not equal) and must consist of
-   * only the characters A-Z, a-z, 0-9, '_', '.', '(', ')'. It must start with
-   * a letter. */
+   * only the characters A-Z, a-z, 0-9, '_', '.', '(', ')', '-', ',', '[', ']'.
+   * It must start with a letter. */
   public final String toString() {
```

Comment:
Why add `-` here ?

Reply:
It also seems that colon ':' is missing in the list.

Reply:
> Why add `-` here ?

'-' is also allowed in pattern , javadoc forgot it

Reply:
> It also seems that colon ':' is missing in the list.

Yes , I have added it. Thanks a a lot.